text/0255-object-safety.md

```diff
@@ -70,63 +70,108 @@ where a trait object is used with a generic call and would be something like
 "type error: SomeTrait does not implement SomeTrait" - no indication that the
 non-object-safe method were to blame, only a failure in trait matching.
 
+Another advantage of this proposal is that it implies that all
+method-calls can always be rewritten into an equivalent [UFCS]
+call. This simplifies the "core language" and makes method dispatch
+notation -- which involves some non-trivial inference -- into a kind
+of "sugar" for the more explicit UFCS notation.
 
 # Detailed design
 
-To be precise about object-safety, an object-safe method:
-* must not have any type parameters,
-* must not take `self` by value,
-* must not use `Self` (in the future, where we allow arbitrary types for the
-  receiver, `Self` may only be used for the type of the receiver and only where
-  we allow `Sized?` types).
+To be precise about object-safety, an object-safe method must meet one
+of the following conditions:
+
+* require `Self : Sized`; or,
+* meet all of the following conditions:
+  * must not have any type parameters; and,
+  * must not take `self` by value; and,
+  * must not use `Self` (in the future, where we allow arbitrary types
+    for the receiver, `Self` may only be used for the type of the
+    receiver and only where we allow `Sized?` types).
```

Comment:
Didn't we also add a restriction making static methods (like the `new` below) non-object safe?